sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -472,6 +473,20 @@ object Overlay {
     builder.append(input.substringSQL(pos + length, Int.MaxValue))
     builder.build()
   }
+
+  def calculate(input: Array[Byte], replace: Array[Byte], pos: Int, len: Int): Array[Byte] = {
+    assert(pos > 0, "If you specify `pos`, it must be a positive whole number.")
```

Comment:
hey.... you cannot do this... Also, can you add tests for negative cases?

Reply:
If pos is a negative whole number, Postgresql, Oracle, DB2, Vertica all think as illegal parameter.